dbms/src/DataStreams/HashJoinBuildBlockInputStream.cpp

```diff
@@ -24,10 +24,17 @@ Block HashJoinBuildBlockInputStream::readImpl()
         Block block = children.back()->read();
         if (!block)
         {
-            join->finishOneBuild();
+            if (join->finishOneBuild(stream_index))
+            {
+                if (join->hasBuildSideMarkedSpillData(stream_index))
```

Comment:
Why the last thread only spill data for `stream_index`, what about other threads that has the remaining blocks that is expected to be spilled?

Reply:
At this time, only stream_index has marked_spill_data, and each thread will call finishOneBuild/finishOneProbe after flushing its own marked_spill_data.